2010-02-21  Michael Tautschnig  <mt@debian.org>

	* setup-storage/Volumes.pm: Handle new mdadm output format (thanks Brian Kroth
		for the patch). (closes: #568794)
Index: trunk/lib/setup-storage/Volumes.pm
===================================================================
--- trunk.orig/lib/setup-storage/Volumes.pm
+++ trunk/lib/setup-storage/Volumes.pm	
@@ -372,7 +372,7 @@
 
   # parse the output line by line
   foreach my $line (@mdadm_print) {
-    if ($line =~ /^ARRAY \/dev\/md(\d+) level=(\S+) num-devices=\d+(\s+|$)/) {
+    if ($line =~ /^ARRAY \/dev\/md[\/]?(\d+) level=(\S+) num-devices=\d+(\s+|$)/) {
       $id = $1;
       $FAI::current_raid_config{$id}{mode} = $2;
       &FAI::push_command( "true", "", "exist_/dev/md$id" );
